lib/pure/collections/sequtils.nim

```diff
@@ -509,11 +509,37 @@ proc keepIf*[T](s: var seq[T], pred: proc(x: T): bool {.closure.})
       inc(pos)
   setLen(s, pos)
 
-func delete*[T](s: var seq[T]; first, last: Natural) =
+func delete*[T](s: var seq[T]; slice: Slice[int]) =
+  ## Deletes the items `s[slice]`, raising a defect if the slice contains
+  ## elements out of range.
+  runnableExamples:
+    var a = @[10, 11, 12, 13, 14]
+    doAssertRaises(AssertionDefect): a.delete(4..5)
+    assert a == @[10, 11, 12, 13, 14]
+    a.delete(4..4)
+    assert a == @[10, 11, 12, 13]
+    a.delete(1..2)
+    assert a == @[10, 13]
+    a.delete(1..<1) # empty slice
+    assert a == @[10, 13]
+  assert slice.a < s.len and slice.a >= 0
+  assert slice.b < s.len and slice.b >= slice.a - 1 # allow 5..4 for empty slice, aka 5..<5
```

Comment:
Allowing `5..4`, but not `5..3` etc. seems like a weird edge case. I'd either not allow `slice.b < slice.a` or allows it and always delete nothing.

Reply:
`n..(n-1)` is special though, it's what you get with empty slices, `n..<n`.
`5..4` makes sense (and is same as `5..<5`, an empty slice) while 5..3 doesn't and indicates a bug somewhere.

Reply:
5..3 is an empty slice just like 5..<5 is and can equally arise when your "step" that you use for iterations is not 1.

Reply:
ok